File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/NearestLandingRegionReply.java
package com.lumiyaviewer.lumiya.slproto.messages;

import com.lumiyaviewer.lumiya.slproto.SLMessage;
import java.nio.ByteBuffer;

/**
 * NearestLandingPointReply
 * dataserver->sim
 * Sent from the data server to a region in reply
 * to the redirectregion request stating which region
 * the requesting region should redirect teleports to if necessary
 *
 * <p>Template: {@code NearestLandingRegionReply Low 145 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
public class NearestLandingRegionReply extends SLMessage {
    public LandingRegionData LandingRegionData_Field;

    /** Block LandingRegionData, Single. */
    public static class LandingRegionData {
        public long RegionHandle; // U64
    }

    public NearestLandingRegionReply() {
        this.zeroCoded = false;
        this.LandingRegionData_Field = new LandingRegionData();
    }

    @Override
    public int CalcPayloadSize() {
        return 12;
    }

    @Override
    public void Handle(SLMessageHandler messageHandler) {
        messageHandler.HandleNearestLandingRegionReply(this);
    }

    @Override
    public void PackPayload(ByteBuffer byteBuffer) {
        // Message number: Low 145 (NearestLandingRegionReply).
        byteBuffer.putShort((short) 0xFFFF);
        byteBuffer.put((byte) 0x00);
        byteBuffer.put((byte) 0x91);
        packLong(byteBuffer, this.LandingRegionData_Field.RegionHandle);
    }

    @Override
    public void UnpackPayload(ByteBuffer byteBuffer) {
        this.LandingRegionData_Field.RegionHandle = unpackLong(byteBuffer);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/messages/NearestLandingRegionReply.kt
package com.lumiyaviewer.lumiya.slproto.messages

import com.lumiyaviewer.lumiya.slproto.SLMessage
import java.nio.ByteBuffer

/**
 * NearestLandingPointReply
 * dataserver->sim
 * Sent from the data server to a region in reply
 * to the redirectregion request stating which region
 * the requesting region should redirect teleports to if necessary
 *
 * <p>Template: {@code NearestLandingRegionReply Low 145 Trusted Unencoded}
 * (recovered/reference/message_template.msg).
 */
open class NearestLandingRegionReply : SLMessage() {
    @JvmField var LandingRegionData_Field: LandingRegionData = LandingRegionData()

    /** Block LandingRegionData, Single. */
    open class LandingRegionData {
        @JvmField var RegionHandle: Long = 0L
    }

    init {
        zeroCoded = false
    }

    override fun CalcPayloadSize(): Int {
        return 12
    }

    override fun Handle(messageHandler: SLMessageHandler) {
        messageHandler.HandleNearestLandingRegionReply(this)
    }

    override fun PackPayload(byteBuffer: ByteBuffer) {
        // Message number: Low 145 (NearestLandingRegionReply).
        byteBuffer.putShort(0xFFFF.toShort())
        byteBuffer.put((0x00).toByte())
        byteBuffer.put((0x91).toByte())
        packLong(byteBuffer, LandingRegionData_Field.RegionHandle)
    }

    override fun UnpackPayload(byteBuffer: ByteBuffer) {
        LandingRegionData_Field.RegionHandle = unpackLong(byteBuffer)
    }
}
